raven_aiohttp.py

```diff
@@ -58,50 +80,129 @@ def keepalive(self):
     def family(self):
         return self._family
 
-    def client_session(self):
-        if not self.keepalive or \
-                (self._client_session is None or self._client_session.closed):
-            connector = aiohttp.TCPConnector(verify_ssl=self.verify_ssl,
-                                             resolve=self.resolve,
-                                             family=self.family,
-                                             loop=self._loop)
-            self._client_session = aiohttp.ClientSession(connector=connector,
-                                                         loop=self._loop)
-        return self._client_session
+    def _client_session_factory(self):
+        # by default aiohttp.TCPConnector deadly caches dns
+        connector = aiohttp.TCPConnector(verify_ssl=self.verify_ssl,
+                                         resolve=self.resolve,
+                                         family=self.family,
+                                         use_dns_cache=False,
+                                         loop=self._loop)
+        return aiohttp.ClientSession(connector=connector,
+                                     loop=self._loop)
+
+    @asyncio.coroutine
+    def close(self, timeout=None, return_exceptions=True):
```

Comment:
Make optional params keyword-only?

Reply:
`return_exceptions` is bad name: `close()` doesn't return anything.
The behavior is much closer to *suppress*.
BTW in what situations suppressing is not desirable?